filestore.go

```diff
@@ -0,0 +1,282 @@
+//go:build experimental
+// +build experimental
+
+// This file declares Go bindings for experimental features in TileDB.
+// Experimental APIs to do not fall under the API compatibility guarantees and
+// might change between TileDB versions
+
+package tiledb
+
+/*
+   	#cgo LDFLAGS: -ltiledb
+   	#cgo linux LDFLAGS: -ldl
+	#include <tiledb/tiledb_experimental.h>
+	#include <tiledb/tiledb_serialization.h>
+	#include <stdlib.h>
+*/
+import "C"
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"runtime"
+	"unsafe"
+)
+
+// CreateFileSchemaFor returns a schema optimized for the storage of the named file.
+// An empty name returns a general schema suitable for any file.
+func CreateFileSchemaFor(tdbCtx *Context, name string) (*ArraySchema, error) {
```

Comment:
maybe CreatedNamedFileSchema and CreateUnnamedFileSchema? Guard checks should probably just be in those functions and share the schema creation logic. Otherwise with this comment I don't see the point of CreateGeneralFileSchema